### ISSUE TITLE
ROL opcode implementation for x86_64 function tracing behaves incorrectly when get_msb() returns >1 bit

### ISSUE THREAD

**Issue by jonbees-ibm** [dod-cyber-crime-center/kordesii#35]
In cases where there are greater than `size` bits in a value, `get_msb()` will return multiple bits. This causes ROL's `opvalue1` on line 1480 of `utils/function_tracing/x86_64/opcodes.py` to contain an incorrect value in some cases. In other places where `get_msb()` is used (e.g. in SAL/SHL) this is addressed by applying the bitmask retrieved by `utils.get_mask(width)` to the result of `get_msb()`, but this is not done for the ROL implementation. 

Adding `opvalue1 &= utils.get_mask(width)` after `opvalue1 = (opvalue1 * 2) + tempcf` on line 1480 of `x86_64/opcodes.py` should address this issue for ROL, but unless `get_msb()` returning multiple bits is assumed in other locations it may make sense to update that function to ensure it always returns a single bit.

Current ROL code (lines 1477-1481), which multiplies `opvalue1` by `2` without applying the width bitmask:
```
if tempcount > 0:
        while tempcount:
            tempcf = get_msb(opvalue1, width)
            opvalue1 = (opvalue1 * 2) + tempcf
            tempcount -= 1
```

An example of updated ROL code which applies the bitmask:
```
if tempcount > 0:
        while tempcount:
            tempcf = get_msb(opvalue1, width)
            opvalue1 = (opvalue1 * 2) + tempcf
            opvalue1 &= utils.get_mask(width)
            tempcount -= 1
```

**Comment by dc3-tsd** [comment]
Thank you for catching this. This is indeed a bug with ROL and it should be masked like you mentioned. We are thinking of including the mask on the ROL implementation like you mentioned as well as have get_msb() throw a an AssertionError if it ends up producing extra bytes due to an incorrectly given size argument. This would help bring any other bugs up to the forefront.

**Comment by dc3-tsd** [comment]
This should be resolved in the 2.5.1 release.
